CONTRIBUTING.md

```diff
@@ -163,3 +163,154 @@ minor versions at this time.
+
+### Dependencies
+
+Sensu uses [golang/dep](https://github.com/golang/dep) for managing its
+dependencies. You will need to install the latest stable version of dep in
+order to install Sensu's dependencies.
+
+[Dep releases](https://github.com/golang/dep/releases)
+
+#### Usage
+
+Running the following will pull all required dependencies, including static
+analysis and linter tools.
+
+```shell
+./build.sh deps
+```
+
+Later, if you would like to add a dependency, run:
+
+```shell
+dep ensure -add https://my-repo.com/my/dep
+```
+
+If you would like to update a dependency, run:
+
+```shell
+dep ensure -update https://my-repo.com/my/dep
+```
+
+When you would like to remove a dependency, remove the it from `Gopkg.toml` and
+then run:
+
+```shell
+dep prune
+```
+
+#### Further Reading
+
+- [The Saga of Go Dependency Management](https://blog.gopheracademy.com/advent-2016/saga-go-dependency-management/)
+- [`dep` Usage](https://github.com/golang/dep#usage)
+
+## Building
+
+### Docker
+
+The simplest way to the build Sensu is with the `sensu-go-build` image. The
+image contains all the required tools to build the agent, backend and sensuctl.
+
+```sh
+docker pull sensu/sensu-go-build
+docker run -it -e GOOS=darwin -v `pwd`:/go/src/github.com/sensu/sensu-go --entrypoint='/go/src/github.com/sensu/sensu-go/build.sh' sensu/sensu-go-build
+```
+
+If you would like to build for different platforms and architectures use GOOS
+and GOARCH env variables. See [Optional environment variables](https://golang.org/doc/install/source#environment) for more.
+
+When complete your binaries will be present in the `target` directory.
+
+### Manually
+
+First ensure that you have the required tools installed to build the programs.
+
+* Ensure that you have the Go tools installed and your environment configured.
+  If not follow the official
+  [Install the Go tools](https://golang.org/doc/install#install) guide.
+* When building the Sensu backend you will need NodeJS and Yarn installed so
+  that the web UI may be included in the binary. Follow
+  [Installing Node.js](https://nodejs.org/en/download/package-manager/) and
+  [Yarn Installation](https://yarnpkg.com/en/docs/install) for installation
+  instructions for your platform.
+
+Once all the tools are installed you are now ready to use the build script. To
+build the Sensu backend, agent and sensuctl, run:
+
+```sh
+./build.sh build
+```
+
+Each product can built separately, with one of the following:
+
+```sh
+./build.sh build_agent
+./build.sh build_backend
+./build.sh build_cli
+```
+
+By default the web UI is built along side and bundled into the backend, as this
+can be a time intensive process, we provide an escape hatch. Use the `dev` tag
+to avoid building the web UI.
+
+```sh
+./build.sh build_backend -tags dev
+```
+
+## Testing
+
+Install dependencies:
+
+```shell
+./build.sh deps
+```
+
+Run test suites:
+
+```shell
+./build.sh ci
```

Comment:
This is actually equivalent to just calling `./build.sh`; `ci` is not a valid sub-command for the script and it will default to doing "everything".